src/pkg/cli/client/byoc/gcp/query.go

```diff
@@ -267,3 +267,11 @@ func (q *Query) AddSince(since time.Time) {
 	}
 	q.baseQuery += fmt.Sprintf(` AND (timestamp >= %q)`, since.UTC().Format(time.RFC3339Nano))
 }
+
+func (q *Query) AddFilter(filter string) {
+	if filter == "" {
+		return
+	}
+	// FIXME: prevent query injection
+	q.baseQuery += fmt.Sprintf(` AND (%v)`, filter)
```

Comment:
Without escaping, this is too brittle. Maybe you can use %q?

Reply:
Please take a look at my comments, %q doesn't prevent injection as it is done via closing brackets.